HABITAT_2_TESTING_GUIDE.md

```diff
@@ -0,0 +1,410 @@
+# Habitat 2.0 Testing Guide
+
+## Overview
+
+This guide provides step-by-step instructions for Testing of Habitat 2.0 RC builds with on-prem-builder. It covers the complete testing workflow from infrastructure setup to custom package development.
+
+
+## Prerequisites
+
+Before starting CFT testing, ensure you have:
+
+- [ ] Linux x86_64 system (minimum: 2 CPUs, 4GB RAM, 100GB disk)
+- [ ] Valid Progress Chef license key for Habitat 2.0
+- [ ] Public builder authentication token
+- [ ] OAuth provider configured (GitHub, Azure AD, etc.)
+- [ ] Network connectivity to bldr.habitat.sh
+
+## Supported Architectures & Channels
+
+**Architectures:**
+- `x86_64-linux` (Intel/AMD 64-bit Linux)
+- `aarch64-linux` (ARM64 Linux)
+- `x86_64-windows` (Intel/AMD 64-bit Windows)
+
+**Channels:**
+- `base` - Core packages for Habitat 2.0
+- `hab-2-rc1` - Chef-specific Habitat 2.0 RC packages
+- `stable` - Habitat 1.x packages (for upgrade testing)
+
+---
+
+## Scenario 1: Setup Standard On-Prem Builder Instance
+
+**Objective:** Setup a standard hab on-prem instance on linux-x86_64
+
+### System Requirements
+
+| Component | Minimum | Recommended |
+|-----------|---------|-------------|
+| CPUs | 2 | 4+ |
+| RAM | 4GB | 8GB+ |
+| Disk | 50GB | 100GB+ |
+| OS | Linux x86_64 with systemd | Ubuntu 20.04+ or RHEL 8+ |
+
+**Reference:** [Detailed system requirements](./docs-chef-io/content/habitat/on_prem_builder/install/system_requirements.md)
+
+### Step 1: Clone Repository
+
+```bash
+git clone https://github.com/habitat-sh/on-prem-builder.git
+cd on-prem-builder
+```
+
+### Step 2: Configure OAuth Provider
+
+Before installation, setup a 3rd-party OAuth provider for authentication:
+
+**Reference:** [OAuth setup guide](./docs-chef-io/content/habitat/on_prem_builder/install/builder_oauth.md#before-you-begin)
+
+**Supported providers:** GitHub, Azure AD, GitLab, Okta, Bitbucket
+
+### Step 3: Configure Environment
+
+```bash
+cp bldr.env.sample bldr.env
+# Edit bldr.env with your OAuth and system configuration
+```
+
+📖 **Reference:** [Configuration guide](./docs-chef-io/content/habitat/on_prem_builder/install/builder_oauth.md#configure-chef-habitat-on-prem-builder)
+
+### Step 4: Install Dependencies
+
+```bash
+# Set authentication token for public builder
+export HAB_AUTH_TOKEN=<PUBLIC_AUTH_TOKEN>
+
+# Install required dependencies
+sudo -E hab pkg install core/sqlite/3130000/20180608141313
+sudo -E hab pkg install core/coreutils/8.32/20240110072150
+```
+
+### Step 5: Run Installation
+
+```bash
+sudo -E ./install.sh
```

Comment:
no need for `sudo -E` here.

Reply:
Removed sudo -E in commit [a117cfd](https://github.com/habitat-sh/on-prem-builder/pull/292/commits/a117cfdc73d5f87c9c1654ccbd1632840d5ec41f)